netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -202,7 +204,14 @@ public long unread() throws IOException {
             )
             .where(DyFriends.HASH, Conditions.equalTo(this.bout))
             .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
+            .iterator();
+        if (!iterator.hasNext()) {
+            throw new RsFailure(
```

Comment:
@2686747 yes exception looks fine now :), but what's with all the other changes in that commit?